saleor/graphql/giftcard/tests/queries/test_gift_card_filtering.py

```diff
@@ -495,3 +496,26 @@ def test_query_filter_gift_cards_by_metadata(
     content = get_graphql_content(response)
     data = content["data"]["giftCards"]["edges"]
     assert len(data) == 1
+
+
+def test_query_filter_gift_cards_by_created_by_email(
+    staff_api_client,
+    gift_card,
+    gift_card_expiry_date,
+    gift_card_used,
+    permission_manage_gift_card,
+):
+    # given
+    query = QUERY_GIFT_CARDS
+
+    variables = {"filter": {"createdByEmail": gift_card.created_by_email}}
+
+    # when
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_gift_card]
+    )
+
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["giftCards"]["edges"]
+    assert data[0]["node"]["createdByEmail"] == "test@example.com"
```

Comment:
Maybe for the sake of this test, we should first update one of the gift cards with some unique email address (in the "given" section) and then expect only one result in the response.

Reply:
@maarcingebala  Even after updating gift_cards getting assert error 
```
gift_card.created_by_email = "random@example.com"
gift_card.save()
```
Because gift card is returning `test@example.com` which is customer email.

Reply:
Maybe we should then reassign the user instance that has the email updated.